scapy/contrib/igmpv3.py

```diff
@@ -66,6 +75,59 @@ def mysummary(self):
     return self.sprintf("IGMPv3 Group Record %IGMPv3gr.type% %IGMPv3gr.maddr%")
 
 
+class IGMPv3Report(Packet):
+  """IGMP Report Class for v3.
+
+  The IGMP v3 Report header differs significantly from other IGMP messages,
+  therefore an specific class is implemented for it.
+
+  'chksum' is automatically calculated before the packet is sent.
+  """
+  name = "IGMPv3Report"
+  fields_desc = [XByteField("type", 0x22),  # Type 0x22 : "V3 Member Report"
+                 ByteField("resv", 0),
+                 XShortField("chksum", None),
+                 ShortField("resv2", 0),
+                 ShortField("numgrprecs", 0)]  # Update manually GR are added
+
+  def igmpize(self, ip=None, ether=None):  # noqa
```

Comment:
you want to overwrite the ip fields as below 

iplong = atol(ip.dst)
ether_dst = "01:00:5e:%02x:%02x:%02x" % ((iplong >> 16) & 0x7F,
                                                                      (iplong >> 8) & 0xFF,
                                                                      (iplong) & 0xFF)
overload_fields = {IP: { "dst": "224.0.0.22", "proto": 2, "ttl": 1, "tos"= 0xc0,
                                    "options"=[IPOption_Router_Alert()] }
                             Ether: {"dst"=ether_dst}
                             }